Sources/Basic/FileSystem.swift

```diff
@@ -467,5 +466,60 @@ public class InMemoryFileSystem: FileSystem {
     }
 }
 
+/// A rerooted view on an existing FileSystem.
+///
+/// This is a simple wrapper which creates a new FileSystem view into a subtree
+/// of an existing filesystem. This is useful for passing to clients which only
+/// need access to a subtree of the filesystem but should otherwise remain
+/// oblivious to its concrete location.
+public struct RerootedFileSystemView: FileSystem {
+    /// The underlying file system.
+    private var underlyingFileSystem: FileSystem
+
+    /// The root path within the containing file system.
+    private let root: AbsolutePath
+    
+    public init(_ underlyingFileSystem: inout FileSystem, rootedAt root: AbsolutePath) {
```

Comment:
It does in fact get changed by the mutating methods (the test case actually uses this).

It would be nice if it could be `inout` only in cases where the client wants a `var`, but the type system can't do this.